### PR TITLE
Add check is_free hook

### PR DESCRIPTION
**Pull request by jaepass**
Added to checkout export, updated README with new hook, built bundle

### CODE REVIEW

File: src/checkout/useCheckIsFree.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  if (!checkout || !commerce) {
+    return null;
+  }
+
+  return commerce.checkout.isFree(
+    checkout.id,
+  ).then((result) => {
+    return result;
+  });
```
Comment (ScopeyNZ) [inline src/checkout/useCheckIsFree.js]:
```suggestion
  useCallback(async () => {
    if (!checkout || !commerce) {
      return null;
    }

    return commerce.checkout.isFree(
      checkout.id,
    ).then((result) => {
      return result;
    });
  }, [checkout, commerce]);
```

BYO import for `useCallback`.

**Review by ScopeyNZ — COMMENTED** [review]
It's probably a mistake that the other checkout callbacks aren't using the `useCallback` hook provided by React. We should aim to use this in new hooks.

File: src/checkout/useCheckIsFree.js (hunk 1 above)
Comment (robbieaverill) [inline src/checkout/useCheckIsFree.js:17]:
```suggestion
  );
```

Comment (jaepass) [reply src/checkout/useCheckIsFree.js]: ok gotcha thanks!

Comment (ScopeyNZ) [reply src/checkout/useCheckIsFree.js]: The `.then` here is also pointless as it's just returning the result.

File: src/checkout/useCheckIsFree.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  useCallback(async () => {
+    if (!checkout || !commerce) {
+      return null;
+    }
+
+    return commerce.checkout.isFree(
+      checkout.id,
+    ).then((result) => {
+      return result;
+    });
```
Comment (robbieaverill) [inline src/checkout/useCheckIsFree.js]:
```suggestion
    );
```

**Comment by jaepass** [comment]
oops fixing

**Review by robbieaverill — APPROVED** [review]

**Pull request merged**